Emby.Server.Implementations/Library/DefaultAuthenticationProvider.cs

```diff
@@ -57,19 +138,26 @@ private bool IsPasswordEmpty(User user, string passwordHash)
 
         public Task ChangePassword(User user, string newPassword)
         {
-            string newPasswordHash = null;
-
-            if (newPassword != null)
+            //string newPasswordHash = null;
+            ConvertPasswordFormat(user);
+            PasswordHash passwordHash = new PasswordHash(user.Password);
+            if(passwordHash.Id == "SHA1" && string.IsNullOrEmpty(passwordHash.Salt))
             {
-                newPasswordHash = GetHashedString(user, newPassword);
+                passwordHash.SaltBytes = _cryptographyProvider.GenerateSalt();
+                passwordHash.Salt = BitConverter.ToString(passwordHash.SaltBytes).Replace("-","");
+                passwordHash.Id = _cryptographyProvider.DefaultHashMethod;
+                passwordHash.Hash = GetHashedStringChangeAuth(newPassword, passwordHash);
+            }else if (newPassword != null)
```

Comment:
```suggestion
            }
            else if (newPassword != null)
```

Reply:
btw, password sent from the UI is always the empty string, so you probably want to use `string.IsNullOrEmpty`?

Reply:
I think I resolved this.